appnexus/client.py

```diff
@@ -84,15 +84,22 @@ def _send(self, send_method, service, data=None, **kwargs):
             logger.debug(' '.join(map(str, (headers, uri, data))))
 
             response = send_method(uri, headers=headers, json=data)
-            response_data = response.json()
             try:
-                self.check_errors(response, response_data["response"])
-            except RateExceeded:
-                self._handle_rate_exceeded(response)
-            except NoAuth:
-                self.update_token()
-            else:
+                response_data = response.json()
+            except ValueError:
+                # Must be a CSV or some other form of data, which is not JSON
                 valid_response = True
+                if response.status_code >= 200:
+                    return response.content
+            else:
+                try:
```

Comment:
Please avoid `try` inside an `else`. I'm pretty sure a simpler structure could be used there, for example by using a bare `raise` if we're not entering into the previous `if`. That would also simplify the diff of that commit.